// collectd - proto/collectd.proto
// Copyright (C) 2015-2016 Sebastian Harl
//
// Permission is hereby granted, free of charge, to any person obtaining a
// copy of this software and associated documentation files (the "Software"),
// to deal in the Software without restriction, including without limitation
// the rights to use, copy, modify, merge, publish, distribute, sublicense,
// and/or sell copies of the Software, and to permit persons to whom the
// Software is furnished to do so, subject to the following conditions:
//
// The above copyright notice and this permission notice shall be included in
// all copies or substantial portions of the Software.
//
// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
// DEALINGS IN THE SOFTWARE.
//
// Authors:
//   Sebastian Harl <sh at tokkee.org>

syntax = "proto3";

package collectd;

import "types.proto";
import "google/protobuf/timestamp.proto";

service Collectd {
	// Dispatch collected values to collectd.
	rpc DispatchValues(DispatchValuesRequest) returns (DispatchValuesReply);

	// Query a list of values available from collectd's value cache.
	rpc QueryValues(QueryValuesRequest) returns (QueryValuesReply);
}

// The arguments to DispatchValues.
message DispatchValuesRequest {
	collectd.types.ValueList values = 1;
}

// The response from DispatchValues.
message DispatchValuesReply {
}

// The arguments to QueryValues.
message QueryValuesRequest {
	// Query by the fields of the identifier. Only return values matching the
	// specified shell wildcard patterns (see fnmatch(3)). Use '*' to match
	// any value.
	collectd.types.Identifier identifier = 1;
}

// The response from QueryValues.
message QueryValuesReply {
	repeated collectd.types.ValueList values = 1;
}
